src/server/endpoints.js

```diff
@@ -277,32 +277,38 @@ function getThreats(gameState, metadata, model) {
 function formatThreats(threats, date) {
   return `Threats ${date}
 =======
+
 ${threats
   .map(
-    (threat, index) => `
-**${index + 1}. ${escapeMarkdownText(threat.title.trim())}**
-${
-  'owner' in threat
-    ? `
-  - *Author:*       ${escapeMarkdownText(threat.owner)}
-`
-    : ''
-}
-  - *Description:*  ${
-    escapeMarkdownText(
+    (threat, index) =>
+      `${index + 1}. **${escapeMarkdownText(threat.title.trim())}**${
```

Comment:
I am fully aware that these nested string-templates are hardly readable. But I also believe adding a proper templating engine as a dependency just for this single function is overkill.

Reply:
Lacking a real templating engine, I think I would find something like the following a lot more readable:
```js
function formatThreat(threat, index) {
  const lines = [
    `${index + 1}. **${escapeMarkdownText(threat.title.trim())}**`,
  ];

  if ('severity' in threat) {
    lines.push(`    - *Severity:* ${escapeMarkdownText(threat.severity)}`);
  }

  if ('owner' in threat) {
    lines.push(`    - *Author:* ${escapeMarkdownText(threat.owner)}`);
  }

  if ('description' in threat) {
    lines.push(
      `    - *Description:* ${escapeMarkdownText(
        threat.description.replace(/(\r|\n)+/gm, ' '),
      )}`,
    );
  }

  if ('mitigation' in threat) {
    lines.push(
      `    - *Mitigation:* ${escapeMarkdownText(
        threat.mitigation.replace(/(\r|\n)+/gm, ' '),
      )}`,
    );
  }

  return lines.join('\n');
}

function formatThreats(threats, date) {
  return `Threats ${date}
=======

${threats.map(formatThreat).join('\n')}
`;
}
```

Reply:
Thank you! I will include this as suggested!